app/ai_backend/test_genesis_api.py

```
@@ -0,0 +1,1359 @@
+    def test_error_context_preservation(self):
+        """Test that error context is preserved for debugging."""
+        error_scenarios = [
+            (401, {"error": "Invalid API key", "error_code": "AUTH_001"}),
+            (429, {"error": "Rate limit exceeded", "retry_after": 60}),
+            (500, {"error": "Internal server error", "trace_id": "abc123"}),
+        ]
+        
+        for status_code, error_data in error_scenarios:
+            try:
+                handle_genesis_error(status_code, error_data)
+                assert False, "Should have raised an exception"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)